deepchem/utils/rdkit_util.py

```diff
@@ -0,0 +1,125 @@
+import logging
+
+import os
+import numpy as np
+import tempfile
+import shutil
+from rdkit import Chem
+from rdkit.Chem import AllChem
+from pdbfixer import PDBFixer
+from simtk.openmm.app import PDBFile
+
+try:
+  from StringIO import StringIO
+except ImportError:
+  from io import StringIO
+
+
+class MoleculeLoadException(Exception):
+
+  def __init__(self, *args, **kwargs):
+    Exception.__init__(*args, **kwargs)
+
+
+def get_xyz_from_mol(mol):
+  """
+  returns an m x 3 np array of 3d coords
+  of given rdkit molecule
+  """
+  xyz = np.zeros((mol.GetNumAtoms(), 3))
+  conf = mol.GetConformer()
+  for i in range(conf.GetNumAtoms()):
+    position = conf.GetAtomPosition(i)
+    xyz[i, 0] = position.x
+    xyz[i, 1] = position.y
+    xyz[i, 2] = position.z
+  return (xyz)
+
+
+def add_hydrogens_to_mol(mol):
+  molecule_file = None
+  try:
+    pdbblock = Chem.MolToPDBBlock(mol)
+    pdb_stringio = StringIO()
+    pdb_stringio.write(pdbblock)
+    pdb_stringio.seek(0)
+    fixer = PDBFixer(pdbfile=pdb_stringio)
+    fixer.addMissingHydrogens(7.4)
```

Comment:
The input mol here is a structure that was generated with removeHs=False, so presumably it has some hydrogens already at some possibly non-physiological pH (eg. 9.0).

But then here we add some additionalHs at a different physiological pH (7.4) ? If so, wouldn't we want to remove all the Hs initially as well (the ones at 9.0)? Shouldn't you have the protonation states of the protein residues be consistent at a given pH?